File: game.js
var min = 1;
var max = 100;
var compGuess;

var attempts = 1;

function getRandInt(min, max){
    return Math.floor(Math.random() * (max - min)) + min;
}

function displayGuess(){
    compGuess = getRandInt(min, max);
    document.getElementById("compGuess").innerHTML = compGuess;
}

function lower(){
    max = compGuess - 1;
    document.getElementById("log").innerHTML += ("IF lower THEN set new max value to old guess. New Max: " + max + "<br>")
    document.getElementById("Max").innerHTML = max
    attempts++;
    displayGuess();
}

function higher(){
    min = compGuess + 1;
    document.getElementById("log").innerHTML += ("IF higher THEN set new min value to old guess. New Min: " + min + "<br>")
    document.getElementById("Min").innerHTML = min
    attempts++;
    displayGuess();
}

function correct(){
    document.getElementById("correct").innerHTML = "Yay! I managed to get it in " + attempts + " tries!"
}

File: game.py
import random

# Initalises values for minimum and maximum to be between 1 and 100
min = 1
max = 100

finished = False

# Initalises a counter for keeping track of attempts
attempts = 0

while finished != True:

    # Computer's choice is a random value between the min and max
    comp_choice = random.randint(min, max)

    # Asks the user if the random number is desired
    print("Is {} your number?".format(comp_choice))
    answer = input("y - Yes\nh - Higher\nl - Lower\n")
    attempts += 1

    # If the number is too low, the program sets the new minimum value to be the choice + 1
    if answer == "h":
        print("\nMy guess of {} was too low, setting new min value to {}".format(comp_choice, comp_choice))
        min = comp_choice + 1

    # If the number is too high, the program sets the new maximum value to be the choice - 1
    elif answer == "l":
        print("\nMy guess of {} was too high, setting new max value to {}".format(comp_choice, comp_choice))
        max = comp_choice - 1
        
    # If the number is correct, the program prints a victory message and shuts down
    elif answer == "y":
        print("Yay I got it in {} tries!".format(attempts))
        finished = True